include/wil/com.h

```diff
@@ -27,2099 +27,2160 @@
+        "comparison operator requires left and right pointers to be compatible");
+    return (left == right.get());
+}
+
+template <typename TLeft, typename TRight, typename ErrRight>
+inline bool operator<(TLeft* left, const com_ptr_t<TRight, ErrRight>& right) WI_NOEXCEPT
+{
+    static_assert(
+        __is_convertible_to(TLeft*, TRight*) || __is_convertible_to(TRight*, TLeft*),
+        "comparison operator requires left and right pointers to be compatible");
+    return (left < right.get());
+}
+
+template <typename TLeft, typename TRight, typename ErrRight>
+inline bool operator!=(TLeft* left, const com_ptr_t<TRight, ErrRight>& right) WI_NOEXCEPT
+{
+    return (!(left == right));
+}
+
+template <typename TLeft, typename TRight, typename ErrRight>
+inline bool operator>=(TLeft* left, const com_ptr_t<TRight, ErrRight>& right) WI_NOEXCEPT
+{
+    return (!(left < right));
+}
+
+template <typename TLeft, typename TRight, typename ErrRight>
+inline bool operator>(TLeft* left, const com_ptr_t<TRight, ErrRight>& right) WI_NOEXCEPT
+{
+    return (right < left);
+}
+
+template <typename TLeft, typename TRight, typename ErrRight>
+inline bool operator<=(TLeft* left, const com_ptr_t<TRight, ErrRight>& right) WI_NOEXCEPT
+{
+    return (!(right < left));
+}
+
+// suppress documentation of every single comparison operator
+/// @endcond
+
+//! An overloaded function that retrieves the raw com pointer from a raw pointer, wil::com_ptr_t<T>, WRL ComPtr<T>, or
+//! Platform::Object^. This function is primarily useful by library or helper code.  It allows code to be written to accept a
+//! forwarding reference template that can be used as an input com pointer.  That input com pointer is allowed to be any of:
+//! * Raw Pointer:  `T* com_raw_ptr(T* ptr)`
+//! * Wil com_ptr:  `T* com_raw_ptr(const wil::com_ptr_t<T, err>& ptr)`
+//! * WRL ComPtr:   `T* com_raw_ptr(const Microsoft::WRL::ComPtr<T>& ptr)`
+//! * C++/CX hat:   `IInspectable* com_raw_ptr(Platform::Object^ ptr)`
+//!
+//! Which in turn allows code like the following to be written:
+//! ~~~~
+//! template <typename U, typename T>
+//! void com_query_to(T&& ptrSource, _COM_Outptr_ U** ptrResult)
+//! {
+//!     auto raw = com_raw_ptr(wistd::forward<T>(ptrSource));
+//!     // decltype(raw) has the type of the inner pointer and raw is guaranteed to be a raw com pointer
+//! ~~~~
+template <typename T>
+T* com_raw_ptr(T* ptr)
+{
+    return ptr;
+}
+
+/// @cond
+template <typename T, typename err>
+T* com_raw_ptr(const wil::com_ptr_t<T, err>& ptr)
+{
+    return ptr.get();
+}
+
+template <typename T>
+T* com_raw_ptr(const Microsoft::WRL::ComPtr<T>& ptr)
+{
+    return ptr.Get();
+}
 
 #ifdef __cplusplus_winrt
 
-    template <typename T>
-    inline IInspectable* com_raw_ptr(T^ ptr)
-    {
-        return reinterpret_cast<IInspectable*>(static_cast<::Platform::Object^>(ptr));
-    }
+template <typename T>
+inline IInspectable* com_raw_ptr(T ^ ptr)

```

Comment:
Hmm. Idk how, if possible at all, to get proper support for C++/CX pointer syntax